table/tables/partition.go

```diff
@@ -129,6 +131,19 @@ func generatePartitionExpr(tblInfo *model.TableInfo) (*PartitionExpr, error) {
 
 		if i > 0 {
 			fmt.Fprintf(&buf, " and ((%s) >= (%s))", pi.Expr, pi.Definitions[i-1].LessThan[0])
+		} else {
+			// NULL will locates in the first partition, so its expression is (expr < value or expr is null).
+			fmt.Fprintf(&buf, " or ((%s) is null)", pi.Expr)
+
+			// Extracts the column of the partition expression, it will be used by partition prunning.
+			if tmp, err1 := expression.ParseSimpleExprWithTableInfo(ctx, pi.Expr, tblInfo); err1 == nil {
```

Comment:
`err1` is not logged.

Reply:
If `expression.ParseSimpleExprWithTableInfo(ctx, pi.Expr, tblInfo)` meets error, 
`expression.ParseSimpleExprWithTableInfo(ctx, pi.Expr < xxx, tblInfo)` will also meets error, and the error will be logged.

The `err1` is not important, we just need to get a `*expression.Column` from a (maybe) string.